### PR TITLE
Implement ord()/chr()

### LINKED ISSUE #5116: ord builtin not supported
## Reporting a bug

- [X] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [X] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

I'm expecting the simple builtin function `ord` to be supported in nopython mode. Instead I get the error `Untyped global name 'ord': cannot determine Numba type of <class 'builtin_function_or_method'>`.

For a running example see https://repl.it/@TomGoBravo/numba-ord

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

Closes #5116

### CODE REVIEW

File: numba/tests/test_unicode.py [hunk 1]
```diff
@@ -2401,5 +2410,60 @@ def test_unicode_literal_stopiteration_iter(self):
                 f()
 
 
+@unittest.skipUnless(_py34_or_later,
```
[the remark this answers is not in the document]
Comment (stuartarchibald) [reply numba/tests/test_unicode.py]: Good point, f9c33fe fixes.

File: numba/tests/test_builtins.py [hunk 2]
```diff
@@ -323,18 +323,6 @@ def test_bool_nonnumber_npm(self):
         with self.assertTypingError():
             self.test_bool_nonnumber(flags=no_pyobj_flags)
 
-    def test_chr(self, flags=enable_pyobj_flags):
```
Comment (esc) [inline numba/tests/test_builtins.py]: Removing this removes all references to the `chr_usecase` defined above. It may well have to be removed also.

File: numba/tests/test_builtins.py [hunk 3]
```diff
@@ -805,18 +793,6 @@ def test_oct_npm(self):
         with self.assertTypingError():
             self.test_oct(flags=no_pyobj_flags)
 
-    def test_ord(self, flags=enable_pyobj_flags):
```
Comment (esc) [inline numba/tests/test_builtins.py]: Removing this also removes all references to the `ord_usecase` defined above. So it may well have to be removed also.

File: numba/unicode.py [hunk 4]
```diff
@@ -2263,6 +2266,50 @@ def _unicode_swapcase(data, length, res, maxchars):
...
+            lc = len(c)
+            if lc != 1:
+                # CPython does TypeError
+                raise TypeError("ord() expected a character")
+            return _get_code_point(c, 0)
+        return impl
+
+
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L2005-L2028    # noqa: E501
+@register_jitable
```
Comment (esc) [inline numba/unicode.py]: This seemingly differs significantly from the cpython implementation referenced above. It might be worthwhile adding a comment or two about it.

File: numba/tests/test_unicode.py [hunk 5]
```diff
@@ -2401,5 +2403,58 @@ def test_unicode_literal_stopiteration_iter(self):
...
+        with self.assertRaises(TypingError) as raises:
+            cfunc(1.23)
+        self.assertIn('Invalid use of Function', str(raises.exception))
+
+    def test_chr(self):
+        pyfunc = chr_usecase
+        cfunc = njit(pyfunc)
+        for ex in UNICODE_EXAMPLES:
+            for x in ex:
+                a = ord(x)
```
Comment (esc) [inline numba/tests/test_unicode.py]: It might be worth adding explicit tests for  the upper and lower boundaries, i.e. `0` and `0x10ffff`.

File: numba/tests/test_builtins.py (hunk 2 above)
Comment (stuartarchibald) [reply numba/tests/test_builtins.py]: I also found a `cmp_usecase` that was dead, removed that too.

File: numba/tests/test_builtins.py (hunk 3 above)
Comment (stuartarchibald) [reply numba/tests/test_builtins.py]: done

File: numba/tests/test_unicode.py (hunk 5 above)
Comment (stuartarchibald) [reply numba/tests/test_unicode.py]: they are covered somewhat by the asserting of invalid values, but its 2 lines so no problem

File: numba/unicode.py [hunk 6]
```diff
@@ -2263,6 +2266,50 @@ def _unicode_swapcase(data, length, res, maxchars):
...
+def _unicode_char(ch):
+    assert ch <= _MAX_UNICODE
+    kind = _codepoint_to_kind(ch)
+    ret = _empty_string(kind, 1, kind == PY_UNICODE_1BYTE_KIND)
```
Comment (stuartarchibald) [inline numba/unicode.py]: bug! Should be `kind==kind`.

Comment (stuartarchibald) [reply numba/unicode.py]: ah, no, is ok, this is just the ascii flag, read that wrong.

File: numba/unicode.py (hunk 4 above)
Comment (stuartarchibald) [reply numba/unicode.py]: It's logically the same, just looks different because the branching etc is abstracted in a different way, I've added a comment to hopefully help.

File: numba/tests/test_builtins.py (hunk 2 above)
Comment (esc) [reply numba/tests/test_builtins.py]: top!

**Comment by stuartarchibald** [comment]
Thanks for the review, marking for merge.
